internal/provider/resource_api_key.go

```diff
@@ -0,0 +1,141 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	neon "github.com/kislerdm/neon-sdk-go"
+)
+
+func resourceAPIKey() *schema.Resource {
+	return &schema.Resource{
+		Description:   "A key to access the Neon API.",
+		SchemaVersion: 7,
+		CreateContext: resourceAPIKeyCreateRetry,
+		ReadContext:   resourceAPIKeyReadRetry,
+		DeleteContext: resourceAPIKeyDeleteRetry,
+		Schema: map[string]*schema.Schema{
+			"key_name": {
```

Comment:
Yeah completely fair. I initially had `key_name` mostly because that is what the parameter is called in [ApiKeyCreateRequest](https://pkg.go.dev/github.com/kislerdm/neon-sdk-go#ApiKeyCreateRequest), but happy to change it, the prefix didn't make me overtly happy either 😄